src/annotator/guest.js

```diff
@@ -247,26 +240,19 @@ export default class Guest extends Delegator {
     });
   }
 
-  addPlugin(name, options) {
-    const Klass = this.options.pluginClasses[name];
-    this.plugins[name] = new Klass(this.element, options, this);
-  }
-
   /**
    * Retrieve metadata for the current document.
    */
   getDocumentInfo() {
     let metadataPromise;
     let uriPromise;
-    if (this.plugins.PDF) {
-      metadataPromise = Promise.resolve(this.plugins.PDF.getMetadata());
-      uriPromise = Promise.resolve(this.plugins.PDF.uri());
-    } else if (this.plugins.Document) {
-      uriPromise = Promise.resolve(this.plugins.Document.uri());
-      metadataPromise = Promise.resolve(this.plugins.Document.metadata);
+
+    if (this.pdfIntegration) {
+      metadataPromise = this.pdfIntegration.getMetadata();
+      uriPromise = this.pdfIntegration.uri();
     } else {
-      uriPromise = Promise.reject();
-      metadataPromise = Promise.reject();
+      uriPromise = Promise.resolve(this.documentMeta.uri());
+      metadataPromise = Promise.resolve(this.documentMeta.metadata);
     }
```

Comment:
The plan in future is that metadata extraction for different document types will have a common interface. In this PR I've just kept the existing structure.